### ISSUE TITLE
[FR] the ability to build a similar interface to the Tasmota Admin interface

### ISSUE THREAD

**Issue by tonyfitzs** [StyraHem/ShellyForHASS#573]

**Comment by maxlyth** [comment]
Take a look at:
https://github.com/maxlyth/hassio-addons/tree/main/shelly-admin

You can add this to Home Assistant via HACS. It exists both as an HA Add-On and also as a stand-alone App for Mac and Windows.

I did a fair bit of work on it at the beginning of the year because I wanted to manage all my Shellys (50+) without depending on a cloud service that I can't control or secure. I asked Shelly support and a few YouTubers what they thought but there was little interest so I refocussed on other projects but could be persuaded back.

**Comment by scooper1** [comment]
Can't get this add-on to show after adding the repository

**Comment by tonyfitzs** [comment]
MAN!!!

Max, this is AWESOME! I don’t know why nobody took it up sooner; I can only suspect that it was because of bandwidth on their part due to other commitments. But truly great, exactly what I was looking for, it shows all my devices and their firmware status. I will play around with it, and if I can think of anything that needs to be updated or changed, I will let you know.

Can I also suggest that this addon is not well published because I looked and looked and could not find anything that even hinted that this existed. This could also explain the lack of interest.

**Comment by scooper1** [comment]
nice interface

**Comment by kongo09** [comment]
Wow, super impressive what @maxlyth has done. Why is that not part of the standard supervisor add ons? I wasn't even aware that you can define additional repositories.

**Comment by maxlyth** [comment]
Thanks for the positive responses. I had a few PMs including a question about password protected devices and this is supported (both on a global password and per-device password level) but not in the release version. 

The Shelly connection library I used did not work for authentication (and other random reliability) so I ended up having to re-write that from the ground up. I applaud anyone that recognises the importance of IoT security but I also realise that this appeals to a minority of the audience so it is only currently in the development branch of the code on Github. More code means potentially new bugs and without testers I did not want to release it.

People have asked why I did not publicise this more but I got the brush-off from Shelly at beta release. I was only after early access to new graphics and meta-data schemas so I could magically make them look great to new customers when they got their shiny new products but I think Shelly want to promote their cloud solution while paying lip-service to open standards, eg MQTT. Don't get me wrong, I have a big investment in Shelly, hence the work on this admin product but it seems the majority think having Amazon listen to their private conversations is acceptable so I can't blame Shelly for going that way.

I use much more recent code for myself but that is to save me time and there have been no Github commits from me for 6 months. I really hoped that there would be interest back in early 2021 but I got distracted on commercial projects. I don't need endorsement from Shelly but public releases are much harder than coding for myself so if anyone can help then I can share what I have.
